### PR TITLE
FMWK-167 Aerospike query creator cleanup

### PR DESCRIPTION
**Pull request by agrgr**

### CODE REVIEW

File: src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java [hunk 1]
```diff
@@ -79,214 +78,92 @@ protected AerospikeCriteria create(Part part, Iterator<Object> iterator) {
 
     private AerospikeCriteria create(Part part, AerospikePersistentProperty property, Iterator<?> parameters) {
         String fieldName = property.getFieldName();
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java:80]: unused variable

File: src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java [hunk 2]
```diff
@@ -79,214 +78,92 @@ protected AerospikeCriteria create(Part part, Iterator<Object> iterator) {
...
-        FilterOperation op;
-        Object v1 = parameters.next(), v2 = null;
+        Object v1 = parameters.next();
         Qualifier.QualifierBuilder qb = new Qualifier.QualifierBuilder();
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java:82]: unused variable

File: src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java [hunk 3]
```diff
@@ -79,214 +78,92 @@ protected AerospikeCriteria create(Part part, Iterator<Object> iterator) {
...
+            case NEGATING_SIMPLE_PROPERTY -> getCriteria(part, property, v1, null, parameters, FilterOperation.NOTEQ);
+            case IN -> getCriteria(part, property, v1, null, parameters, FilterOperation.IN);
+            default -> throw new IllegalArgumentException("Unsupported keyword '" + part.getType() + "'");
+        };
+    }
+
+    public AerospikeCriteria getCriteria(Part part, AerospikePersistentProperty property, Object v1, Object v2,
+                                Iterator<?> parameters, FilterOperation op) {
+        Qualifier.QualifierBuilder qb = new Qualifier.QualifierBuilder();
+        String fieldName = part.getProperty().getSegment(); // Map bin name, later passed to Exp.mapBin());
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java:116]: redundant closing bracket and semicolon in the comment

File: src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java [hunk 4]
```diff
@@ -79,214 +78,92 @@ protected AerospikeCriteria create(Part part, Iterator<Object> iterator) {
...
+        String dotPath = null;
+        Object v3 = null;
+
+        if (property.isCollectionLike()) {
+            op = getCorrespondingListFilterOperationOrFail(op);
+        } else if (property.isMap()) {
+            List<Object> params = new ArrayList<>();
+            parameters.forEachRemaining(params::add);
+            Object nextParam = params.get(0);
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java:125]: should go under the `if (params.size() == 1)`

**Review by reugn — APPROVED** [review]

**Review by roimenashe — APPROVED** [review]

**Pull request merged**